### ISSUE TITLE
Reading File with UTF-8 BOM produce Syntax Error

### ISSUE THREAD

**Issue by aersam** [tconbeer/sqlfmt#381]
**Describe the bug**
When calling sqlfmt on a file starting the famous UTF-8 BOM, it produces an error:
```
 sqlfmt encountered an error: Could not parse SQL at position 1: '»¿SELECT
```

**Expected behavior**
Well, it should just read that file. And also emit in UTF-8-BOM if easily possible

**Additional context**
What is the output of `sqlfmt --version`?
`sqlfmt, version 0.16.0`

**Comment by tconbeer** [comment]
Thanks for the report, @aersam. We've been punting on supporting other encodings (and I'll include windows-flavored utf-8 in that), but it's probably about time to fix that.

**Comment by aersam** [comment]
Thanks a lot! Looks like something python should have built-in if you ask me

**Comment by tconbeer** [comment]
Yeah, the file munging stuff is frustratingly hard to get right, especially across OSes. 

This fix has been released in v0.17.0
